text_mining.py

```diff
@@ -0,0 +1,217 @@
+"""Mines text from Project Gutenburg. Performs word frequency and sentiment analysis
+on text and generates a word cloud from most common words.
+
+@author: Vivien Chen
+
+"""
+
+import requests
+import string
+import math
+from pickle import dump, load
+from os.path import exists
+from stop_words import get_stop_words       # pip install stop_words
+from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
+from wordcloud import WordCloud             # pip install wordcloud
+
+
+def get_cache(url, file_name):
+    """Writes text from url into file_name and returns text if file_name does not exist.
+    Otherwise, simply returns text from file_name. Assumes that text from file_name
+    matches text from url if file_name exists.
+
+    Args:
+        url: url you want to read text from
+        file_name: name of file you want to write to/read from
+
+    Returns:
+        text from url
+    """
+    if exists(file_name) == False:
+        file_ = open(file_name, 'wb')
+        text = requests.get(url).text
+        dump(text, file_)
+
+    return open(file_name, 'r', encoding='utf-8', errors='ignore')
+
+
+def filter_PG_text(text):
+    """Takes the raw text of a Project Gutenberg book as input. Strips away header
+    comments and returns the book portion of the text.
+
+    Args:
+        text: the raw text from a Project Gutenberg book
+
+    Returns:
+        the book portion of the Project Gutenberg book
+    """
+    lines = text.readlines()
+
+    start_line = 0
+    while lines[start_line].find('START OF THIS PROJECT GUTENBERG EBOOK') == -1:
+        start_line += 1
+    lines = lines[start_line+1:]
+
+    end_line = 0
+    while lines[end_line].find('END OF THIS PROJECT GUTENBERG EBOOK') == -1:
+        end_line += 1
+    lines = lines[:end_line-3]
+
+    return ' '.join(lines)
+
+
+def get_word_list(text):
+    """Takes a string of text as input. Strips away punctuation and whitespace.
+    Converts all words to lowercase. Returns a list of the words used in the string
+    with stop words removed.
+
+    Args:
+        text: a string of text, such as the filtered text from a Project Gutenberg book
+
+    Returns:
+        a list of words, all lowercase, from the string with stop words removed
+    """
+    text = text.lower()
+    word_list = text.split()
+
+    for i in range(len(word_list)):
+        word_list[i] = word_list[i].strip(string.punctuation)
+
+    stop_words = get_stop_words('en')
+    stop_words_2 = ["a", "about", "above", "across", "after", "afterwards", "again", "against", "all",
```

Comment:
Out of curiosity, what happens if you only use the stop_words words, rather than your manually-assembled ones?

Reply:
stop_words = ['a', 'about', 'above', 'after', 'again', 'against', 'all', 'am', 'an', 'and', 'any', 'are', "aren't", 'as', 'at', 'be', 'because', 'been', 'before', 'being', 'below', 'between', 'both', 'but', 'by', "can't", 'cannot', 'could', "couldn't", 'did', "didn't", 'do', 'does', "doesn't", 'doing', "don't", 'down', 'during', 'each', 'few', 'for', 'from', 'further', 'had', "hadn't", 'has', "hasn't", 'have', "haven't", 'having', 'he', "he'd", "he'll", "he's", 'her', 'here', "here's", 'hers', 'herself', 'him', 'himself', 'his', 'how', "how's", 'i', "i'd", "i'll", "i'm", "i've", 'if', 'in', 'into', 'is', "isn't", 'it', "it's", 'its', 'itself', "let's", 'me', 'more', 'most', "mustn't", 'my', 'myself', 'no', 'nor', 'not', 'of', 'off', 'on', 'once', 'only', 'or', 'other', 'ought', 'our', 'ours', 'ourselves', 'out', 'over', 'own', 'same', "shan't", 'she', "she'd", "she'll", "she's", 'should', "shouldn't", 'so', 'some', 'such', 'than', 'that', "that's", 'the', 'their', 'theirs', 'them', 'themselves', 'then', 'there', "there's", 'these', 'they', "they'd", "they'll", "they're", "they've", 'this', 'those', 'through', 'to', 'too', 'under', 'until', 'up', 'very', 'was', "wasn't", 'we', "we'd", "we'll", "we're", "we've", 'were', "weren't", 'what', "what's", 'when', "when's", 'where', "where's", 'which', 'while', 'who', "who's", 'whom', 'why', "why's", 'with', "won't", 'would', "wouldn't", 'you', "you'd", "you'll", "you're", "you've", 'your', 'yours', 'yourself', 'yourselves']

Reply:
It worked pretty well, but there were a few common words that weren't included in stop_words (I can only remember 'one' being a very common word off the top of my head), so I just googled another set of stop words and copied it over.